### PR TITLE
chore: add codeowners file

### LINKED ISSUE #1089: 📎 Setup CODEOWNERS
Can be used for PR review recommendations and allow people to opt in to parts of the codebase they want to be responsible for or care about.

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes #1089 

I filled the file with the latest information left in that PR, plus I added the new core contributors based on their last area of work. 

I'd like to have some feedback about the format. Later on, we could add/modify sections based on our needs.

@yeonjuan @jer3m01 @sebmck could you please review it?

## Test Plan

CI

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/internal/html-parser/* @ematipico
+
+/internal/commit-parser/* @ematipico
+
+/internal/code-js-manifest/* @ematipico
+
+/internal/browser-features/* @jer3m01
+
+/internal/browser-db/* @jer3m01
```
Comment (jer3m01) [inline CODEOWNERS:21]: Can also add me to `codec-browsers`

**Review by jer3m01 — CHANGES_REQUESTED** [review]
Good for me, one to add.

**Comment by sebmck** [comment]
Looks good to me! Why so many newlines though?

**Comment by ematipico** [comment]
> Looks good to me! Why so many newlines though?

Thought it was more readable. Just changed it

**Comment by ematipico** [comment]
The paths are not in alphabetical order

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# The next sections address specific people that help to maintain particular packages inside the project
+
+* @sebmck
+
+/internal/browser-features/* @jer3m01
+/internal/browser-db/* @jer3m01
+/internal/commit-parser/* @ematipico
+/internal/compiler/lint/* @ematipico @bitpshr
+/internal/code-browsers/* @jer3m01
+/internal/code-js-manifest/* @ematipico
```
Comment (jer3m01) [inline CODEOWNERS:14]: Forgot a `c` on codec.

**Review by jer3m01 — APPROVED** [review]
Great! Just one small typo but everything looks good.

File: CODEOWNERS (hunk 2 above)
Comment (sebmck) [inline CODEOWNERS:14]:
```suggestion
/internal/codec-js-manifest/* @ematipico
```

**Comment by sebmck** [comment]
Thanks @ematipico! Anyone feel free to PR any changes, including removing or adding yourself wherever.

**Pull request merged**